src/lib/schema/SchemaEntry.ts

```diff
@@ -0,0 +1,202 @@
+	 */
+	public default: SerializableValue;
+
+	/**
+	 * The minimum value for this entry.
+	 */
+	public minimum: number | null;
+
+	/**
+	 * The maximum value for this entry.
+	 */
+	public maximum: number | null;
+
+	/**
+	 * Whether this entry should inclusively or exclusively check minimum and maximum on key validation.
+	 */
+	public inclusive: boolean;
+
+	/**
+	 * Whether or not this entry should be configurable by the configuration command.
+	 */
+	public configurable: boolean;
+
+	/**
+	 * The filter to use for this entry when resolving.
+	 */
+	public filter: SchemaEntryFilterFunction | null;
+
+	/**
+	 * Whether or not values managed by this entry should be resolved.
+	 */
+	public shouldResolve: boolean;
+
+	public constructor(parent: Schema | SchemaFolder, key: string, type: string, options: SchemaEntryOptions = {}) {
+		this.client = null;
+		this.parent = parent;
+		this.key = key;
+		this.path = this.parent.path.length === 0 ? this.key : `${this.parent.path}.${this.key}`;
+		this.type = type.toLowerCase();
+		this.array = typeof options.array === 'undefined' ? typeof options.default === 'undefined' ? false : Array.isArray(options.default) : options.array;
+		this.default = typeof options.default === 'undefined' ? this.generateDefaultValue() : options.default;
+		this.minimum = typeof options.minimum === 'undefined' ? null : options.minimum;
+		this.maximum = typeof options.maximum === 'undefined' ? null : options.maximum;
+		this.inclusive = typeof options.inclusive === 'undefined' ? false : options.inclusive;
+		this.configurable = typeof options.configurable === 'undefined' ? this.type !== 'any' : options.configurable;
+		this.filter = typeof options.filter === 'undefined' ? null : options.filter;
+		this.shouldResolve = typeof options.resolve === 'undefined' ? true : options.resolve;
+	}
+
+	/**
+	 * Get the serializer that manages this instance.
+	 */
+	public get serializer(): Serializer | null {
+		if (this.client === null) throw new Error('Cannot retrieve serializers from non-initialized SchemaEntry.');
+		return this.client.serializers.get(this.type) || null;
+	}
+
+	/**
+	 * Edits this SchemaEntry instance.
+	 * @param options The options to edit
+	 */
+	public edit(options: SchemaEntryEditOptions = {}): this {
+		if (typeof options.type === 'string') this.type = options.type.toLowerCase();
+		if (typeof options.array !== 'undefined') this.array = options.array;
+		if (typeof options.configurable !== 'undefined') this.configurable = options.configurable;
+		if (typeof options.default !== 'undefined') this.default = options.default;
+		if (typeof options.filter !== 'undefined') this.filter = options.filter;
+		if (typeof options.inclusive !== 'undefined') this.inclusive = options.inclusive;
+		if (typeof options.resolve !== 'undefined') this.shouldResolve = options.resolve;
+
+		if (('minimum' in options) || ('maximum' in options)) {
```

Comment:
are the additional braces required here? IIRC no?

Reply:
They might turn the code into `if ('minimum' in (options || boolean))` so, yes

Reply:
Yes.